tests/test_pipeline.py

```diff
@@ -275,6 +316,90 @@ def test_gridsearchcv(self) -> None:
             for k, value in param_grid.items():
                 self.assertIn(grid_search_cv.best_params_[k], value)
 
+    def test_caching(self) -> None:
+        """Test if the caching gives the same results and is faster on the second run."""
+
+        molecule_net_logd_df = pd.read_csv(
+            "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/Lipophilicity.csv"
+        ).head(1000)
+        pipeline = _get_rf_regressor()
+        mem = Memory(location="./cache", verbose=0)
```

Comment:
Will this directory be cleared after the test run and is this platform independent? I think creating a temporary directory with https://docs.python.org/3/library/tempfile.html would be the better choice here.

Reply:
Done